pages/contact/forms.py

```diff
@@ -0,0 +1,181 @@
+	"""
+
+	name = forms.CharField(min_length=2, max_length=100, required=False)
+	email = forms.EmailField(required=False)
+	message = forms.CharField(min_length=20, max_length=5000, widget=forms.Textarea)
+	suggestion = forms.BooleanField(required=False)
+	dm_support = forms.BooleanField(
+		required=False, label="Request for data management/sharing support"
+	)
+	other = forms.BooleanField(required=False)
+
+	# Anti-spam fields
+	website = forms.CharField(required=False, widget=forms.HiddenInput)
+	contact_ts = forms.CharField(widget=forms.HiddenInput, strip=False)
+	contact_dsc = forms.CharField(widget=forms.HiddenInput, strip=False)
+
+	# Internal state for logging (not exposed to users)
+	_blocked_reason: Optional[str] = None
+
+	def __init__(self, *args: Any, request=None, **kwargs: Any) -> None:
+		"""Initialise the form.
+
+		Args:
+			*args: Positional form args.
+			request: Optional HttpRequest to access cookies for token checks.
+			**kwargs: Keyword form args.
+		"""
+		super().__init__(*args, **kwargs)
+		self.request = request
+
+	def clean_email(self) -> str:
+		"""Validate email and prevent header injection.
+
+		Returns:
+			The sanitised email value.
+
+		Raises:
+			ValidationError: If CR/LF characters are present.
+		"""
+		value = self.cleaned_data.get("email", "")
+		if "\r" in value or "\n" in value:
+			raise ValidationError("Enter a valid email address.")
+		return value
+
+	def clean_message(self) -> str:
+		"""Validate message content against HTML and URL constraints.
+
+		Returns:
+			The validated message content.
+
+		Raises:
+			ValidationError: If HTML-like tags are present or URL count exceeds
+			    the allowed threshold.
+		"""
+		value = self.cleaned_data.get("message", "")
+		# Reject HTML-like content
+		if "<" in value or ">" in value:
+			raise ValidationError("Please remove HTML tags.")
+		# Cap URLs
+		url_count = len(URL_REGEX.findall(value))
+		if url_count > MAX_URLS_ALLOWED:
+			raise ValidationError("Too many links in the message.")
+		return value
+
+	def clean(self) -> dict[str, Any]:
+		"""Form-level validation for category, timing, and token checks.
+
+		Returns:
+			The cleaned data dictionary.
+
+		Raises:
+			ValidationError: On any anti-spam or category selection failure.
+		"""
+		cleaned = super().clean()
+
+		# At least one category must be selected
+		if not (
+			cleaned.get("suggestion")
+			or cleaned.get("dm_support")
+			or cleaned.get("other")
+		):
+			raise ValidationError("Please select at least one option.")
+
+		# Honeypot
+		if cleaned.get("website"):
+			self._blocked_reason = "HONEYPOT_HIT"
+			raise ValidationError(
+				"We couldn't submit the form. Please try again in a moment."
+			)
+
+		# Timing token
+		ts_token = cleaned.get("contact_ts") or ""
+		try:
+			# Accept TimestampSigner style token or JSON payload signed via dumps
+			try:
+				signer = signing.TimestampSigner(salt="contact-ts")
+				ts_str = signer.unsign(ts_token, max_age=MAX_TOKEN_AGE_SECONDS)
```

Comment:
We had a discussion about having a upper limit check while [changing data platform's contact form](https://github.com/ScilifelabDataCentre/data.scilifelab.se/pull/428#discussion_r2191967509), and I still stand by it. 

Persons like me might get easily distracted and comeback after 1 hour, might type a long message and submit, only to fail. That could be really frustrating 😅 So unless there is a notification after 60 saying to reload the page, which prevents the user from typing, we should not have upper limit for time check.

Reply:
Yes I remember :) I put it again so I don't forget to raise it during the discussion with everyone

Reply:
Ok :)